src/charm.py

```diff
@@ -349,7 +349,14 @@ def _on_peer_relation_joined(self, event: RelationJoinedEvent):
 
         # Add new instance to the cluster
         try:
-            self._mysql.add_instance_to_cluster(new_instance_fqdn, new_instance_label)
+            cluster_primary = self._mysql.get_cluster_primary_address()
+            if not cluster_primary:
+                self.unit.status = BlockedStatus("Unable to retrieve the cluster primary")
```

Comment:
you mentioned this was an issue in the PR description, why? Wouldn't we want to try again and defer?

Reply:
here, i think it is better to return after moving the unit into blocked status because we have confirmed that the instance is configured for the innodb cluster, but we are unable to get the cluster primary's address. this usually will indicate a problem to be resolved which we wont surface if we defer